### PR TITLE
Cypress test to log in with Login.gov

### LINKED ISSUE #1552: Can log in with a Login.gov user on staging

### PR DESCRIPTION
**Pull request by neilmb**
This adds a single Cypress test that can log in to our app with Login.gov. It also sets it up to run in Github Actions against our staging environment as part of the daily "deploy staging" workflow. Further work after this will expand the login test into a complete audit submission (see #1399 for more details).

There are lots of people who could review this: @eazapata23 since he made it all happen, @heymatthenry and @jperson1 since they know Cypress, and @asteel-gsa who did the Github Actions setup.

Closes #1552.

### CODE REVIEW

**Review by asteel-gsa — APPROVED** [review]
Approving the actions.

As a note, when this comes to other envs, we need to have an input for `environment` like some of the other files do.
```
    with:
      environment: "staging"
```
simply so we can add the hooks into the respective deployment workflows.

Modified:
- removed e2e tests from running
- modified `on` triggers

File: docs/testing.md [hunk 1]
```diff
@@ -115,3 +116,36 @@ You can also run most of them as one command with `make lint`.
...
+
+We use Cypress to do end-to-end testing of the application. Tests are defined
+in files in [backend/cypress/e2e/](/backend/cypress/e2e).
+
+## Testing behind Login.gov
+
+ For tests that run against our existing Cloud.gov environments, we can't use
+ our `DISABLE_AUTH` environment variable to allow the end-to-end tests access
+ to our authenticated pages. Instead, we need to use an actual Login.gov
+ sadnbox account to log in.
```
Comment (jperson1) [inline docs/testing.md:129]: `sadnbox` -> `sandbox`

File: docs/testing.md [hunk 2]
```diff
@@ -115,3 +116,36 @@ You can also run most of them as one command with `make lint`.
...
+ Google Group, create a random password and record it locally, ideally in a
+ KeePassXC password vault. When the account creation flow asks to set up a
+ second factor, choose "Authentication Application". On the next screen where
+ you would normally use a QR code to configure an authenticator application,
+ there is a text box with a secret key labeled "enter this code manually into
+ your authentication app". You need to save that secret key alongside the
+ password.
+
+ To pass that email address, password, and secret key into Cypress, set the
+ environment variables `CYPRESS_LOGIN_EMAIL`, `CYPRESS_LOGIN_PASSWORD`, and
```
Comment (jperson1) [inline docs/testing.md:147]: `CYPRESS_LOGIN_EMAIL`, `CYPRESS_LOGIN_PASSWORD`, `CYPRESS_LOGIN_OTP_SECRET`
->
`CYPRESS_LOGIN_TEST_EMAIL`, `CYPRESS_LOGIN_TEST_PASSWORD`, `CYPRESS_LOGIN_TEST_OTP_SECRET`

**Review by jperson1 — APPROVED** [review]
Looks great, and works well for me! Approving now, but requesting minor documentation changes.

**Pull request merged**